test/bcreader_tests.jl

```diff
@@ -0,0 +1,257 @@
+                    ),],
+                ),
+            )
+            datafile_rll = joinpath(sst_data, "sst.nc")
+            varname = "SST"
+
+            regrid_dir = "bcreader_regrid_dir"
+            isdir(regrid_dir) ? nothing : mkpath(regrid_dir)
+
+            bcf_info = BCReader.bcfile_info_init(
+                FT,
+                regrid_dir,
+                datafile_rll,
+                varname,
+                boundary_space_t,
+                segment_idx0 = [Int(1309)],
+                interpolate_daily = false,
+                land_mask = land_mask_t,
+            )
+
+            dates = (; date = [date], date0 = [date0], date1 = [date1])
+            cs_t = (;
+                _info = bcf_info,
+                dates = dates,
+                SST_all = [],
+                updating_dates = [],
+                monthly_state_diags = (; fields = deepcopy(dummy_data), ct = [0]),
+            )
+
+            # step in time
+            walltime = @elapsed for t in ((tspan[1] + Δt):Δt:tspan[end])
+                cs_t.dates.date[1] = CallbackManager.current_date(cs_t, t) # if not global, `date`` is not updated. Check that this still runs when distributed.
+
+                model_date = cs_t.dates.date[1]
+                callback_date = BCReader.next_date_in_file(cs_t._info)
+
+                # TODO investigate if macro would be faster here
+                if (model_date >= callback_date)
+                    BCReader.update_midmonth_data!(model_date, cs_t._info)
+                    push!(cs_t.SST_all, deepcopy(cs_t._info.monthly_fields[1]))
+                    push!(cs_t.updating_dates, deepcopy(model_date))
+                end
+
+            end
+
+            # test if the SST field was modified
+            @test cs_t.SST_all[end] !== cs_t.SST_all[end - 1]
+            # check that the final file date is as expected
+            @test Date(cs_t.updating_dates[end]) == Date(1979, 03, 16)
+
+            rm(regrid_dir; recursive = true, force = true)
+        end
+
+        @testset "test bcf_info_init for FT=$FT" begin
+            # setup for test
+            radius = FT(6731e3)
+            Nq = 4
+            domain = Domains.SphereDomain(radius)
+            mesh = Meshes.EquiangularCubedSphere(domain, 4)
+            topology = Topologies.Topology2D(mesh)
+            quad = Spaces.Quadratures.GLL{Nq}()
+            boundary_space_t = Spaces.SpectralElementSpace2D(topology, quad)
+            land_mask_t = ClimaCore.Fields.zeros(boundary_space_t)
+
+            # import data from file
+            sst_data = AW.get_data_folder(
+                AW.ArtifactWrapper(
+                    @__DIR__,
+                    "sst",
+                    AW.ArtifactFile[AW.ArtifactFile(
+                        url = "https://caltech.box.com/shared/static/8gd3wjq2dbrnzuv8pd5ww3a54h0kkcz8.nc",
+                        filename = "sst.nc",
+                    ),],
+                ),
+            )
```

Comment:
Note that this can lead to a race condition (if this same code is exercised in different buildkite jobs), since artifacts are saved to a common path. Please see [this PR](https://github.com/CliMA/ClimaAtmos.jl/pull/981) for an example of how this race condition can be avoided.